### PR TITLE
[chore][exporter/debug] refactor code to make independent from Logging exporter

### PR DESCRIPTION
**Pull request by astencel-sumo**
This PR is the first part of this draft PR: https://github.com/open-telemetry/opentelemetry-collector/pull/9298.

This refactoring is needed to implement [[exporter/debug] change behavior of "normal" verbosity to be different from "basic" #7806](https://github.com/open-telemetry/opentelemetry-collector/issues/7806). I want to change the behavior of the Debug exporter, but leave the behavior of the Logging exporter unchanged.

**Link to tracking Issue:**

- https://github.com/open-telemetry/opentelemetry-collector/issues/7806

### CODE REVIEW

**Review by mx-psi — COMMENTED** [review]
No strong opinion here, but an alternative solution could be to have a custom unmarshal function/initialization logic for the logging exporter that overrides verbosity from `normal` to set to `basic`. That way we would avoid duplicating code. Any thoughts on which one is better?

**Comment by astencel-sumo** [comment]
> No strong opinion here, but an alternative solution could be to have a custom unmarshal function/initialization logic for the logging exporter that overrides verbosity from `normal` to set to `basic`. That way we would avoid duplicating code. Any thoughts on which one is better?

I feel we might have more changes coming to the Debug exporter that I don't think we want to introduce to the Logging exporter - see e.g. https://github.com/open-telemetry/opentelemetry-collector/issues/9372, https://github.com/open-telemetry/opentelemetry-collector/issues/9921. I would leave the Logging exporter's code unchanged, waiting to be removed in September 2024, and only work with the Debug exporter's code.

**Comment by mx-psi** [comment]
Okay, I am fine with this approach, let me ping @open-telemetry/collector-approvers just in case someone has a different opinion and we can wait a couple days to give people time to voice their opinion

File: exporter/debugexporter/exporter.go
Comment (mx-psi) [inline exporter/debugexporter/exporter.go:3]: Can we add a comment here and in the original copy of this code that states:
- where the other copy of this codes live
- that, for security issues, we should update both

?

Comment (astencel-sumo) [reply exporter/debugexporter/exporter.go:3]: I'm not sure it's that important, as these files don't have much logic to begin with, but I've added the following: https://github.com/open-telemetry/opentelemetry-collector/pull/9922/commits/8dce26b26d3d5d8fd5514552747d0c084b5fa504.

**Review by mx-psi — APPROVED** [review]
Will merge tomorrow unless somebody else objects

**Pull request merged**
